### DISCUSSION TITLE
Hot water charge

### DISCUSSION THREAD

**Discussion by MichalAug** [Blackymas/NSPanel_HA_Blueprint#469] [Q&A]
In the thermostat menu, it is possible to add a "hot water charge" switch.
I wanted to modify it a bit, because in the case of the viessmann boiler it is not a switch, but a button. Cannot be disabled from the HA level.

button.activate_one_time_charge

I wanted it to turn off after pressing this icon.

Where to look for it and how does it work if in esphome.yaml if this part is commented out?

 ##### Hotwater Charge #####
 # - platform: nextion
 # name: ${device_name} Hotwater Charge
 # page_id: 6
 # component_id: 20
     # on_click:
     # - homeassistant.service:
     # service: switch.toggle
     # date:
     # entity_id: ${ha_hotwater_charge}

**Comment by hollabee** [comment]
Just for my understanding: The Viessmann Boiler is exposed to HA, and you can switch the hot water charge on and off using HA? But the switch in the display should be turning off after let's say 2 seconds?

You could do so by creating a helper:
- Create a helper button called input_boolean.viesmann_toggle
- Create an automation with:
- trigger "if input_boolean.viesmann_toggle changes to on"
- action "whatever Viessmann is doing"
- Delay for 5 Seconds
- Input boolean: Toggle your input_boolean.viesmann_toggle

In the Blueprint, for the hot water charge, you call the helper.

Would that help?

**Reply by MichalAug** [reply]
Thank you. I'll have to think about it calmly and try to do as you described.
It works a bit strange, because from the HA level I can only turn on "hot water charge". I can't turn off the heating from the HA level.
That's why I want a monostable button. I will try the way you described.

**Comment by edwardtfn** [comment]
Is this still a need with the new layout of climate page?
